File: src/main/java/com/medicalsystem/service/impl/ExcelServiceImpl.java
package com.medicalsystem.service.impl;

import com.medicalsystem.excel.exporter.ExcelExporter;
import com.medicalsystem.excel.importer.ExcelImporter;
import com.medicalsystem.service.ExcelService;
import lombok.AllArgsConstructor;
import lombok.extern.java.Log;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.InputStreamResource;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.*;

@Service
@AllArgsConstructor(onConstructor = @__(@Autowired))
@Log
public class ExcelServiceImpl implements ExcelService {

    private final String IMPORT_FILE_NAME = "data/dataImport.xlsx";
    private final String EXPORT_FILE_NAME = "data/dataExport.xlsx";

    private final ExcelExporter exporter;
    private final ExcelImporter importer;

    @Override
    public ResponseEntity<?> exportToFile() {
        ResponseEntity<?> response;
        try {
            exporter.exportToFile(EXPORT_FILE_NAME);
            File file = new File(EXPORT_FILE_NAME);
            InputStreamResource resource = new InputStreamResource(new FileInputStream(file));
            response = ResponseEntity.ok()
                    .contentLength(file.length())
                    .contentType(MediaType.parseMediaType("application/octet-stream"))
                    .body(resource);
        } catch (IOException e) {
            log.severe(e.getMessage());
            response = new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
        }
        return response;
    }

    @Override
    public ResponseEntity<?> importToDatabase(MultipartFile file) {
        ResponseEntity<?> response;
        try {
            saveFile(file);
            importer.importToDatabase(IMPORT_FILE_NAME);
            response = new ResponseEntity<>(HttpStatus.OK);
        } catch (IOException e) {
            log.severe(e.getMessage());
            response = new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
        }
        return response;
    }

    private void saveFile(MultipartFile file) throws IOException {
        if (file.isEmpty()) {
            throw new IOException("The file is empty");
        }

        try (BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(new File(IMPORT_FILE_NAME)))) {
            byte[] bytes = file.getBytes();
            stream.write(bytes);
            log.info("File uploaded: " + IMPORT_FILE_NAME);
        }
    }
}

File: src/main/kotlin/com/medicalsystem/service/impl/ExcelServiceImpl.kt
package com.medicalsystem.service.impl

import com.medicalsystem.excel.exporter.ExcelExporter
import com.medicalsystem.excel.importer.WorkbookImporter
import com.medicalsystem.service.ExcelService
import com.medicalsystem.util.ExcelUtils
import com.medicalsystem.util.FileUtils
import org.apache.poi.ss.usermodel.Workbook
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import org.springframework.web.multipart.MultipartFile
import java.io.IOException

@Service
class ExcelServiceImpl @Autowired constructor(
        val workbookImporter: WorkbookImporter,
        val excelExporter: ExcelExporter) : ExcelService {

    private val IMPORT_FILE_PATH = "data/dataImport.xlsx"
    private val EXPORT_FILE_PATH = "data/dataExport.xlsx"

    override fun importToDatabase(file: MultipartFile): ResponseEntity<*> =
            try {
                FileUtils.saveFile(file, IMPORT_FILE_PATH)
                importToDatabase(IMPORT_FILE_PATH)
                ResponseEntity.ok().body("File imported")
            } catch (e: IOException) {
                ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.message)
            }

    override fun exportToFile(): ResponseEntity<*> {
        TODO("not implemented")
    }

    override fun importToDatabase(path: String) {
        val workbook: Workbook? = ExcelUtils.loadWorkbook(path)
        workbook?.let {
            workbookImporter.importToDatabase(workbook)
        }
    }

    override fun exportToFile(path: String) =
            excelExporter.exportToFile(path)
}